This should fix the Exception partially

MaSven opened the pull request:
This will be hopefully the last big PR, to fix #18 .

MaSven committed:
Fix Exceptions partially

Signed-off-by: MaSven <dev@mail.smarquardt.space>

MaSven committed:
Fix Exceptions partially

Signed-off-by: MaSven <dev@mail.smarquardt.space>

MaSven committed:
we should throw here

Because the read failed here, it is better to rethrow it

Signed-off-by: MaSven <dev@mail.smarquardt.space>

MaSven committed:
autostyle apply

Signed-off-by: MaSven <dev@mail.smarquardt.space>

MaSven committed:
reused name clashed with other variable

Signed-off-by: MaSven <dev@mail.smarquardt.space>

weisJ commented on src/main/java/BasisStatus/StatusIdle.java:
```diff
@@ -905,7 +905,7 @@ public void mouseMoved(MouseEvent e) {
             try {
                 vmObject.processAddElement(mainPath, circuitClass, frontClass, args);
-            } catch (Exception ex) {
+            } /* TODO why the exception here? */ catch (Exception ex) {
```
Maybe it calls functions on elements. Because users can add own element implementations (I think) we need to make sure no exception is thrown. But this assumption probably has too much faith in whoever write this.

weisJ commented on src/main/java/MyParser/Parser.java:
```diff
@@ -103,12 +103,12 @@ private boolean isIn(String ch, String set) {
      * ch=val.substring(i,i+1); if (isIn(ch,ALPHA)==false) return false; } return true; }
      */
-
+    // TODO NumberUtils anyone?
```
Rather no NumberUtils but an api that is enough strongly typed to not check.

MaSven replied:
Something in that direction. I saw this code 5 times in a row. Where something is checked to be a number.

weisJ commented on src/main/java/SimpleFileSystem/Main.java:
```diff
@@ -41,28 +42,24 @@ public Main() {}
     public static void main(String[] args) {
```
I thing this class can be removed complete it doesn't seem to be used anywhere.

weisJ replied:
By thing I mean the whole class

MaSven replied:
addresed here 067d2b19

weisJ commented on src/main/java/VisualLogic/ElementPalette.java:
```diff
@@ -160,16 +152,20 @@ public String getEndung(String fileName) {
+
+        } catch (FileNotFoundException e) {
+            org.tinylog.Logger.error(e);
+        } catch (IOException ioException) {
```
Why not combine both exceptions into one path?

MaSven replied:
addresed here 0aaf39a8

weisJ commented on src/main/java/VisualLogic/StoreImageException.java:
```diff
@@ -0,0 +1,40 @@
+
+import java.io.IOException;
+
+public class StoreImageException extends IOException {
```
Please move new exception classes into the `com.github.mylibrelab` package. (Same applies for the exceptions from the last PR).

MaSven replied:
Fixed here abb21125

weisJ commented on src/main/java/VisualLogic/variables/VS1DInteger.java:
```diff
@@ -74,30 +77,32 @@ public void copyValueFrom(Object in) {
-        } catch (Exception ex) {
+        } catch (IOException ex) {
+            Logger.error(ex);
             System.err.println("Fehler in VS1DInteger.saveToStream() : " + ex.toString());
```
`System.err` call should be removed.

MaSven replied:
Should we fix this in this PR too?

weisJ replied:
Depending on how many occurrences we are speaking of this may be a separate PR.

MaSven replied:
```bash
❯ rg --count "System.err.println" src/ | cut -d':' -f 2 | count
16
```

weisJ replied:
I thinks that is manageable to include in here.

MaSven replied:
alright

weisJ commented on src/main/java/VisualLogic/variables/VS1DString.java:
```diff
@@ -78,30 +81,32 @@ public void copyValueFrom(Object in) {
-        } catch (Exception ex) {
+        } catch (IOException ex) {
+            Logger.error(ex);
             System.err.println("Fehler in VS1DString.loadFromStream() : " + ex.toString());
```
`System.err` call should be removed. It might be worth while to search for all uses of it. Won't comment on other occurrences.

weisJ commented on src/main/java/VisualLogic/variables/VSImage.java:
```diff
@@ -116,13 +115,13 @@ public void loadFromStream(java.io.FileInputStream fis) {
                     image.flush();
                 }
-
+                // TODO ignored read here
```
Is this still Todo?

MaSven replied:
Yes just wanted to point it out. Sometimes reads from a stream are just ignored.

weisJ replied:
Ok could you then clarify this in the comment a bit more? I though this was a Todo related to this PR.

MaSven replied:
Added comment here 1e91c2fd

MaSven committed:
moved exceptions to mylibrelab package

Signed-off-by: MaSven <dev@mail.smarquardt.space>

MaSven committed:
collapse exceptions

Signed-off-by: MaSven <dev@mail.smarquardt.space>

MaSven committed:
delete not used class

Signed-off-by: MaSven <dev@mail.smarquardt.space>

MaSven committed:
addres todo in comment

I leave this comment here because either
* A
We don't need to read here and this is code left from some other purpose
* B
This adresses some bug where the image must be read first before awt can
to something with the bytes. Or this is to check the image can be read
from.

Signed-off-by: MaSven <dev@mail.smarquardt.space>

MaSven committed:
reimport moved exceptions

Signed-off-by: MaSven <dev@mail.smarquardt.space>

MaSven commented:
> Ah ok now I see what's happening here. `FileInputStream#read` populates the content of the byte array.

Yes you are right. It saves the bytes into the imageBytes array. I will delete the comment. Totally missed that

marko-radosavljevic commented:
Please, before merging this, check other occurrences of bad exception handling with tools like SonarCloud, SpotBugs and PMD.

It would be nice if we could deal with all the bad handling in this PR, so we don't have to do these global exception fixes again. ^^
[Filtered exception bugs.](https://sonarcloud.io/project/issues?id=MyLibreLab_key254582&resolved=false&severities=BLOCKER&types=BUG)

MaSven committed:
Revert "addres todo in comment"

This reverts commit 1e91c2fd12dd669f7aa37127e5428af5600115d1.

MaSven committed:
delete unneeded comment

Signed-off-by: MaSven <dev@mail.smarquardt.space>

MaSven commented:
> Please, before merging this, check other occurrences of bad exception handling with tools like SonarCloud, SpotBugs and PMD.
> 
> I would be nice if we could deal with all bad handling in this PR, so we don't have to do these global exception fixes again. ^^
> [Filtered exception bugs.](https://sonarcloud.io/project/issues?id=MyLibreLab_key254582&resolved=false&severities=BLOCKER&types=BUG)

We have another big PR. If I change too much here the other PR will get problems merging them. What does @weisJ think?

weisJ commented:
From the legacy files my PR actually only touches `DialogLicence` and `FrameMain` and basically all changes are deletions . Everything else is self contained and shouldn't make any problems with the changes here.

MaSven commented:
Ok i will check sonar if there is anything left for Exceptionhandling.

MaSven committed:
Partially fix Exceptions

Signed-off-by: MaSven <dev@mail.smarquardt.space>

MaSven committed:
autostyle

Signed-off-by: MaSven <dev@mail.smarquardt.space>

MaSven committed:
fix exception partially

Signed-off-by: MaSven <dev@mail.smarquardt.space>

MaSven committed:
fix exception partially

Signed-off-by: MaSven <dev@mail.smarquardt.space>

MaSven commented:
So i think i now have everything. Let me know if there is anything left.

MaSven committed:
fix exception partially

Signed-off-by: MaSven <dev@mail.smarquardt.space>

MaSven commented on src/main/java/de/myopenlab/update/UnzipFiles.java:
```diff
@@ -48,29 +48,31 @@ public void unzip(String zipFilePath, String destDirectory) throws IOException {
 
-                String tmp = new File(filePath).getParent();
-                new File(tmp).mkdirs();
+                String filePath = destDirectory + File.separator + entry.getName();
```
Yes we write here Arbitrary. But that should not matter right now.

Commit:
Update FileSystemInput.java

Removed one `System.out.println()`
